lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -926,6 +929,84 @@ static void checkOverrideAccessControl(ValueDecl *baseDecl, ValueDecl *decl,
   }
 }
 
+static GenericSignature *getNewGenericSignature(ValueDecl *base,
```

Comment:
This looks good. Can you try using this new function in CodeSynthesis.cpp as well?

Reply:
@slavapestov So sorry for the late reply (wwdc had me distracted)!

I can, where do you want me to use this method in CodeSynthesis?

Reply:
I was thinking it would be used by configureGenericDesignatedInitOverride(). Does that make sense? It looks like it's almost the same computation.

Reply:
@slavapestov Yeah it would be great to get this PR merged first and then I can clean up CodeSynthesis and do the vtable check thing in separate PRs :)